sdk/python/tests/dsl/metadata_tests.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from kfp.dsl._metadata import ComponentMeta, ParameterMeta, TypeMeta
+import unittest
+
+class TestTypeMeta(unittest.TestCase):
+  def test_from_dict(self):
+    component_dict = {
+        'GCSPath': {
+            'bucket_type': 'directory',
+            'file_type': 'csv'
+        }
+    }
+    golden_type_meta = TypeMeta(name='GCSPath', properties={'bucket_type': 'directory',
+                                                          'file_type': 'csv'})
+    self.assertEqual(TypeMeta.from_dict(component_dict), golden_type_meta)
+
+  def test_eq(self):
+    type_a = TypeMeta(name='GCSPath', properties={'bucket_type': 'directory',
+                                                  'file_type': 'csv'})
+    type_b = TypeMeta(name='GCSPath', properties={'bucket_type': 'directory',
+                                                  'file_type': 'tsv'})
+    type_c = TypeMeta(name='GCSPatha', properties={'bucket_type': 'directory',
+                                                  'file_type': 'csv'})
+    type_d = TypeMeta(name='GCSPath', properties={'bucket_type': 'directory',
+                                                  'file_type': 'csv'})
+    self.assertNotEqual(type_a, type_b)
+    self.assertNotEqual(type_a, type_c)
+    self.assertEqual(type_a, type_d)
+
+
+class TestComponentMeta(unittest.TestCase):
+
+  def test_to_dict(self):
+    component_meta = ComponentMeta(name='foobar',
+                                   description='foobar example',
+                                   inputs=[ParameterMeta(name='input1',
+                                                         description='input1 desc',
+                                                         param_type=TypeMeta(name='GCSPath',
+                                                                             properties={'bucket_type': 'directory',
+                                                                                         'file_type': 'csv'
+                                                                                         }
+                                                                             ),
+                                                         default='default1'
+                                                         ),
+                                           ParameterMeta(name='input2',
```

Comment:
Can you also test the `Integer` type serialization?

Reply:
Done.